### PR TITLE
fix: don't enable Sentry if DSN is not defined

### PR DESCRIPTION
**Pull request by paulschreiber**
## Description
Don't enable Sentry if DSN is not defined. Matches backend behaviour.

### CODE REVIEW

File: dev-client/src/App.tsx [hunk 1]
```diff
@@ -46,12 +46,16 @@ import * as Sentry from '@sentry/react-native';
 import {enableFreeze} from 'react-native-screens';
 import {HomeScreenContextProvider} from 'terraso-mobile-client/screens/HomeScreen/HomeScreen';
 
+const sentryEnabled = APP_CONFIG.sentryDsn;
```
Comment (shrouxm) [inline dev-client/src/App.tsx:49]: nit:

```suggestion
const sentryEnabled = Boolean(APP_CONFIG.sentryDsn);
```

**Review by shrouxm — COMMENTED** [review]
questions: would it be better to just require sentry dsn to be defined, do we ever want to not have sentry enabled? i've found it useful even when doing local dev. and if we do want to disable it, could we have a separate required config flag `sentryEnabled`? with the change as is, if we make a build configuration error and sentryDsn is not set we will silently not have sentry enabled in any new builds. that seems like a more likely build configuration error to make than accidentally explicitly setting `sentryEnabled` to `false`

**Comment by paulschreiber** [comment]
@garo If someone who is not Tech Matters runs Terraso, they may not have Sentry installed/configured. I want the app to be able to launch successfully for them.

**Comment by shrouxm** [comment]
@paulschreiber that makes sense, in that case i have the same question

> could we have a separate required boolean config flag sentryEnabled? with the change as is, if we make a build configuration error and sentryDsn is not set we will silently not have sentry enabled in any new builds. that seems like a more likely build configuration error to make than accidentally explicitly setting sentryEnabled to false

File: dev-client/.env.sample [hunk 2]
```diff
@@ -7,3 +7,4 @@ PUBLIC_MAPBOX_TOKEN=""
 SENTRY_DSN=https://xyz@o4506299756642304.ingest.sentry.io/abcd
 SENTRY_DISABLE_AUTO_UPLOAD="true"
 SENTRY_PROJECT="terraso-lpks"
+SENTRY_ENABLED="false"
```
Comment (shrouxm) [inline dev-client/.env.sample:10]: question: why are some of these quoted and some not?

File: dev-client/src/App.tsx [hunk 3]
```diff
@@ -46,12 +46,16 @@ import * as Sentry from '@sentry/react-native';
 import {enableFreeze} from 'react-native-screens';
 import {HomeScreenContextProvider} from 'terraso-mobile-client/screens/HomeScreen/HomeScreen';
 
+const sentryEnabled = Boolean(APP_CONFIG.sentryEnabled);
```
Comment (shrouxm) [inline dev-client/src/App.tsx:49]: nit: the `Boolean` call is redundant

File: dev-client/src/config/index.ts [hunk 4]
```diff
@@ -99,6 +100,10 @@ if (Config.SENTRY_DSN === undefined) {
   throw new Error('Config setting SENTRY_DSN not set');
 }
 
+if (Config.SENTRY_ENABLED === undefined) {
+  throw new Error('Config setting SENTRY_ENABLED not set');
+}
```
Comment (shrouxm) [inline dev-client/src/config/index.ts:105]: nit:
```suggestion
if (Config.SENTRY_ENABLED !== 'true' && Config.SENTRY_ENABLED !== 'false' ) {
  throw new Error(`Config setting SENTRY_ENABLED set to invalid value: ${Config.SENTRY_ENABLED}`);
}
```

**Review by shrouxm — APPROVED** [review]
couple non-blocking nits/questions

File: dev-client/.env.sample [hunk 5]
```diff
@@ -7,3 +7,4 @@ PUBLIC_MAPBOX_TOKEN=""
 SENTRY_DSN=https://xyz@o4506299756642304.ingest.sentry.io/abcd
```
Comment (paulschreiber) [inline dev-client/.env.sample:7]:
```suggestion
SENTRY_DSN="https://xyz@o4506299756642304.ingest.sentry.io/abcd"
```

File: dev-client/.env.sample [hunk 6]
```diff
@@ -4,6 +4,7 @@ GOOGLE_OAUTH_IOS_CLIENT_ID="abcdefghijklmnopqrstuvwxyz.apps.googleusercontent.co
 TERRASO_BACKEND=https://api.staging.terraso.net
```
Comment (paulschreiber) [inline dev-client/.env.sample:4]:
```suggestion
TERRASO_BACKEND="https://api.staging.terraso.net"
```

File: dev-client/.env.sample [hunk 7]
```diff
@@ -1,9 +1,10 @@
 ENV=development
```
Comment (paulschreiber) [inline dev-client/.env.sample:1]:
```suggestion
ENV="development"
```

File: dev-client/.env.sample (hunk 2 above)
Comment (paulschreiber) [reply dev-client/.env.sample:10]: no idea. i quoted all of them for consistency.

**Pull request merged**